src/obr/signac_wrapper/operations.py

```diff
@@ -55,6 +67,40 @@ def set_entrypoint(self, entrypoint: dict):
         """
         self._entrypoint = entrypoint
 
+    def group_jobs(self, jobs: list[Job], summarize: int = 0) -> dict[str, list[Job]]:
+        """Returns the list of jobs of the given OpenFOAMProject where the last `summarize` levels are grouped together at the corresponding parent view."""
+        grouped = {}
+        id_view_map = map_view_folder_to_job_id(os.path.join(self.path, "view"))
+        for job in jobs:
+            jobid = str(job.id)
+            # only consider leaf nodes for now
+            if summarize and job.statepoint["has_child"]:
+                continue
+            current = dict(job.statepoint)
+            p_view = None
+            # follow parent links "bottom-up" in statepoint <summarize> times
+            pid = (
+                calc_nth_parent(to_go=summarize, parent=current) if summarize else jobid
+            )
+            if summarize and pid is None:
+                # if pid is None, that means that the chosen summarize value was larger than the tree level of job
+                continue
+            if summarize and pid:
+                p_view = id_view_map.get(
+                    pid, ""
+                )  # default shouldn't be needed, but just to be safe
+                p_view = p_view.replace(pid, "")  # remove pid from end of view
+            if p_view and p_view not in grouped:
+                grouped[p_view] = []
+            view = p_view or id_view_map.get(jobid)
+            job.doc["state"]["view"] = view
```

Comment:
```suggestion
```
I think that should be somewhere else, otherwise it an unexpected side effect and hard to find later.